tests/DocumentationExamplesTest.php

```diff
@@ -1180,6 +1184,223 @@ public function testIndex_example_2()
+                throw $error;
+            }
+        }
+    }
+    // End Transactions Retry Example 2
+
+    // Start Transactions Retry Example 3
+    private function runTransactionWithRetry3(callable $txnFunc, Manager $manager, Session $session)
+    {
+        while (true) {
+            try {
+                $txnFunc($manager, $session);  // performs transaction
+                break;
+            } catch (\MongoDB\Driver\Exception\CommandException $error) {
+                $resultDoc = $error->getResultDocument();
+
+                // If transient error, retry the whole transaction
+                if (isset($resultDoc->errorLabels) && in_array('TransientTransactionError', $resultDoc->errorLabels)) {
+                    continue;
+                } else {
+                    throw $error;
+                }
+            } catch (\MongoDB\Driver\Exception\Exception $error) {
+                throw $error;
+            }
+        }
+    }
+
+    private function commitWithRetry3(Manager $manager, Session $session)
+    {
+        while (true) {
+            try {
+                $session->commitTransaction();
+                echo "Transaction committed.\n";
+                break;
+            } catch (\MongoDB\Driver\Exception\CommandException $error) {
+                $resultDoc = $error->getResultDocument();
+
+                if (isset($resultDoc->errorLabels) && in_array('UnknownTransactionCommitResult', $resultDoc->errorLabels)) {
+                    echo "UnknownTransactionCommitResult, retrying commit operation ...\n";
+                    continue;
+                } else {
+                    echo "Error during commit ...\n";
+                    throw $error;
+                }
+            } catch (\MongoDB\Driver\Exception\Exception $error) {
+                echo "Error during commit ...\n";
+                throw $error;
+            }
+        }
+    }
+
+    private function updateEmployeeInfo3(Manager $manager, Session $session)
+    {
+        $employees = (new Database($manager, "hr"))->employees;
+        $events = (new Database($manager, "reporting"))->events;
+
+        $session->startTransaction([
+            'readConcern' => new ReadConcern("snapshot"),
+            'writeConcern' => new WriteConcern(WriteConcern::MAJORITY)
+        ]);
+
+        try {
+            $employees->updateOne(
+                ['employee' => 3],
+                ['$set' => ['status' => 'Inactive']],
+                ['session' => $session]
+            );
+            $events->insertOne(
+                ['employee' => 3, 'status' => [ 'new' => 'Inactive', 'old' => 'Active']],
+                ['session' => $session]
+            );
+        } catch (\MongoDB\Driver\Exception\Exception $error) {
+            echo "Caught exception during transaction, aborting.\n";
+            $session->abortTransaction();
+            throw $error;
+        }
+
+        $this->commitWithRetry3($manager, $session);
```

Comment:
`commitWithRetry3()` shouldn't require the `$manager` (or `$db` if you make other suggested changes).

Reply:
Done